File: MyApps/EarthquakeViewer/app/src/main/java/com/example/luispena/earthquakeviewer/MainActivity.java
package com.example.luispena.earthquakeviewer;

import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.example.luispena.earthquakeviewer.Fragments.EarthQuakeListFragment;

public class MainActivity extends AppCompatActivity {

    private static final String TAG_LIST_FRAGMENT = "TAG_LIST_FRAGMENT";
    EarthQuakeListFragment mEarthquakeListFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        FragmentManager fm = getSupportFragmentManager();

        if (savedInstanceState == null){

            FragmentTransaction ft = fm.beginTransaction();

            mEarthquakeListFragment = new EarthQuakeListFragment();
            ft.add(R.id.main_activity_frame, mEarthquakeListFragment,TAG_LIST_FRAGMENT);
            ft.commitNow();
        }else{
            mEarthquakeListFragment = (EarthQuakeListFragment)fm.findFragmentByTag(TAG_LIST_FRAGMENT);
        }


    }
}

File: KotlinCodeLabs/DiceRollerImage - 01.3/app/src/main/java/com/example/diceroller/MainActivity.kt
package com.example.diceroller

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {
    /*
    The lateinit keyword promises the Kotlin compiler that the variable will be initialized before the code calls any operations on it.
    Therefore we don't need to initialize the variable to null here, and we can treat it as a non-nullable variable when we use it.
    It is a best practice to use lateinit with fields that hold views in just this way. */
    lateinit var diceImage : ImageView


    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        diceImage = findViewById(R.id.dice_image)
        roll_button.setOnClickListener{ roll()}



    }

    private fun roll(){

        val randomInt = Random().nextInt(6) + 1

        // With android kotlin extensions I do not have to set FindViewByID I can just get the ID
        // through the xml.
        //val diceImage: ImageView = findViewById(R.id.dice_image)

        val drawableResource = when(randomInt){
            1 -> R.drawable.dice_1
            2 -> R.drawable.dice_2
            3 -> R.drawable.dice_3
            4 -> R.drawable.dice_4
            5 -> R.drawable.dice_5
            else-> R.drawable.dice_6
        }

        diceImage.setImageResource(drawableResource)


    }





}
